keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -231,15 +238,34 @@ class ApplicationService(
     }
 
     val artifactSummaries = deliveryConfig.artifacts.map { artifact ->
-      val artifactVersionSummaries = repository.artifactVersions(artifact, limit).map { artifactVersion ->
+      val artifactVersions = repository.artifactVersions(artifact, limit)
+
+      // A verification context identifies an artifact version in an environment
+      // For each context, there may be multiple verifications (e.g., test-container, canary)
+      //
+      // This map associates a context with this collection of verifications and their states
+      val vStateMap = getVerificationStates(deliveryConfig, artifactVersions)
```

Comment:
Consider renaming the variable with a meaningful name:
```suggestion
      val verificationStateMap = getVerificationStates(deliveryConfig, artifactVersions)
```